File: ScaleformUI_Csharp/Items/UIMenuSeparatorItem.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ScaleformUI
{
	public class UIMenuSeparatorItem : UIMenuItem
	{
		public bool Jumpable = false;
		/// <summary>
		/// Use it to create an Empty item to separate menu Items
		/// </summary>
		public UIMenuSeparatorItem(string title, bool jumpable) : base(title, "")
		{
			_itemId = 6;
			Jumpable = jumpable;
		}

		public override void SetLeftBadge(BadgeIcon badge)
		{
			throw new Exception("UIMenuSeparatorItem cannot have a left badge.");
		}
		public override void SetRightBadge(BadgeIcon badge)
		{
			throw new Exception("UIMenuSeparatorItem cannot have a right badge.");
		}
		public override void SetRightLabel(string text)
		{
			throw new Exception("UIMenuSeparatorItem cannot have a right label.");
		}
	}
}

File: ScaleformUI_V8/src/menus/UIMenu/items/uimenuseparatoritem.ts
import { BadgeStyle } from "elements/badge";
import { UIMenuItem } from "./uimenuitem";

export class UIMenuSeparatorItem extends UIMenuItem{
    public Jumpable:boolean;
    constructor(title:string, jumpable:boolean){
        super(title, "");
        this.Jumpable = jumpable;
        this._itemId = 6;
    }
    
    public SetLeftBadge(badge: BadgeStyle) {
        throw new Error("UIMenuSeparatorItem cannot have a left badge.");
    }

    public SetRightBadge(badge: BadgeStyle) {
        throw new Error("UIMenuSeparatorItem cannot have a right badge.");
    }
    
    public SetRightLabel(text: string) {
        throw new Error("UIMenuSeparatorItem cannot have a right label.");
    }
}